dist/api_data.py

```diff
@@ -1034,6 +1034,10 @@ def __ge__(self, other):
             @ref tune_durability for more information''',
             choices=['dsync', 'fsync', 'none']),
         ]),
+    Config('verify_metadata', 'false', r'''
+        open connection and verify any WiredTiger-owned table. This API
+        allows verification and detection of corruption in WiredTiger tables.''',
```

Comment:
Does calling `wiredtiger_open` with this flag set result in a valid open connection once verify has completed?

I know why we got here, but I still think a method on the `WT_CONNECTION` is a more natural API for users. How much harder is that to implement?

Reply:
I'll defer to @sueloverso, but I think it's going to be quite difficult.

We'd have to somehow split apart the existing `wiredtiger_open()` handling of these files, as it opens and reads them, and we'd also have to guarantee no other methods were called between the `wiredtiger_open()` call and the new verify method call, which implies some kind of locking.

What are the semantics if `wiredtiger_open()` succeeds in opening these files, but then they fail verify in the new method?

I could be totally missing an approach (??), but it seems hard to me.

Reply:
I agree with @keithbostic. We have to verify before opening the history store or metadata file. We cannot verify them after they're open. That is the whole problem with what exists today. That is why it is strategically called before the metadata is opened the first time. We don't really have a useful connection if the metadata and history store don't verify.